pallets/creditcoin/src/lib.rs

```diff
@@ -968,94 +969,56 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[transactional]
```

Comment:
transactional!

Reply:
Should we keep the use of [transactional] as a separate issue? If we wish to start using it, we should probably do it globally.

Reply:
Since these are new functions and we know we will be going this way I'm ok with them being in this PR. We should make a ticket to update all the extrinsics though.

Reply:
What happens with events triggered midexecution if the call fails in the end?

Reply:
Events are really just backed by storage (events are a storage item in the system pallet) so any events emitted are reverted on failure as well